src/main/java/chatlinitation/ChatViewmodel.java

```diff
@@ -0,0 +1,44 @@
+package chatlinitation;
+
+import entities.PrivateChat;
+
+/**
+ * chatviewmodel  connects our ChatView UI to our Entities.
+ * It provide data to the view, so that view can put that data on the screen.
+ * containing the message content and associated metadata
+ * this class contain privatechat, recipeint user and Textmessge.
+ */
+
+public class ChatViewmodel {
+
+    PrivateChat privatechat;
+    User recipient;
+    TextMessage txtmessage;
+
```

Comment:
This is related to my next comment; It might be preferable to not reference entities on the outer layers. Instead, we can pass information to the inner layers for them to handle. For the user-typed message content/string as an example, this string should be passed to the interactor, who will create a new TextMessage.

Reply:
yea good idea I already add some interactors instead